blockly-core/core/block.js

```diff
@@ -598,16 +602,20 @@ Blockly.Block.prototype.onMouseUp_ = function(e) {
       }
       inferiorConnection.sourceBlock_.svg_.connectionUiEffect();
     }
-    if (this.blockSpace.trashcan && this.blockSpace.trashcan.isOpen) {
+    if (thisBlockSpace.trashcan) {
       // Don't throw an object in the trash can if it just got connected.
-      this.blockSpace.trashcan.close();
-    }
-  } else if (this.blockSpace.trashcan && this.blockSpace.trashcan.isOpen) {
-    var trashcan = this.blockSpace.trashcan;
-    goog.Timer.callOnce(trashcan.close, 100, trashcan);
-    if (Blockly.selected) {
-      Blockly.selected.dispose(false, true);
-    }
+      thisBlockSpace.trashcan.close();
+    }
+  } else if (Blockly.selected &&
+             Blockly.selected.isDeletable() &&
+             thisBlockSpace.isDeleteArea(e)) {
```

Comment:
I think our preferred style is to indent these lines by 2 or 4 spaces, rather than all the way to the parens.